### PR TITLE
Ia color input only

### PR DESCRIPTION
**Pull request by ianyimi**
color input and analytics fix to block local changes from posthog

### CODE REVIEW

File: src/realities/BuilderReality/tools/BuilderWand/index.tsx [hunk 1]
```diff
@@ -33,6 +33,14 @@ export function useBuilderWand() {
...
 }
 
+const ColorsContext = createContext<string[]>(
+  []
+)
+
+export function useColorHistory() {
+  return useContext(ColorsContext)
+}
+
```
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/index.tsx:43]: looks like this isn't being used?

File: src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/components/SchemaGenerator.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ export function SchemaGenerator(props: { ideaProps: Record<string, unknown> }) {
   const { idea } = useFocus();
 
   const [value, setValue] = useState<string>("");
+  const [color, setColor] = useState<string>("");
```
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/components/SchemaGenerator.tsx:13]: how come you need a separate state? seems like you can just use value

File: src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/ColorInput.tsx [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+              <animated.meshBasicMaterial color={pastColor} />
+            </mesh>
+          </FocusInteractable>
+          <FocusInteractable
+            onHover={() => {
+              setHover1(true);
+            }}
+            onUnHover={() => {
+              setHover1(false);
+            }}
```
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/ColorInput.tsx:348]: this file needs to be cleaned up so it can quickly be edited (look at Rotate/index.tsx as an example). recommend turning all this (couldn't select all of it) into a separate component

File: src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/ColorInput.tsx [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+  const { history, imagination } = useDream();
+  const { scene } = useThree();
+
+  const inputRef = useRef<HTMLInputElement>();
+  const [focused, setFocused] = useState(false);
+  const [match, setMatch] = useState<string>("");
+  const [inputHover, setIHover] = useState<boolean>(false);
+  const [hover, setHover] = useState<boolean>(false);
+  const [hover1, setHover1] = useState<boolean>(false);
+  const [hover2, setHover2] = useState<boolean>(false);
```
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/ColorInput.tsx:54]: numbered variables is a huge sign that a component can be created

File: src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/ColorInput.tsx [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+import { FocusInteractable } from "realities/BuilderReality/modifiers/FocusInteractable";
+
+type ColorProps = {
+  value: string;
+  setValue: (s: string) => void;
+  attribute: string;
+  ideaProps: Record<string, unknown>;
+  enabled?: boolean;
+  inputType?: "text" | "password" | "email";
+  autoComplete?: boolean;
```
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/ColorInput.tsx:24]: some of this doesn't seem relevant to color input.. if we're going to keep copying and pasting text input code we should abstract it, at least locally to the builder reality. or just clean up some of the irrelevant code

File: src/realities/BuilderReality/tools/BuilderWand/index.tsx (hunk 1 above)
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/index.tsx:43]: think this can move to the prema panel

also if you're going to use it you need to include 
`<ColorsContext.Provider value={...}>
     {children}
</ColorsContext.Provider>`

or else it's not going to work

**Review by alex-shortt — CHANGES_REQUESTED** [review]
`src/layers/basis/index.tsx` doesn't need to be renamed

File: src/realities/BuilderReality/tools/BuilderWand/index.tsx (hunk 1 above)
Comment (ianyimi) [reply src/realities/BuilderReality/tools/BuilderWand/index.tsx:43]: moved to prema panel index and seems to be working without tags

File: src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/index.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
 export * from "./MediaInput";
 export * from "./TextInput";
-export * from "./BoolInput";
\ No newline at end of file
+export * from "./BoolInput";
+export * from "./ColorInput/index";
```
Comment (alex-shortt) [inline src/realities/BuilderReality/tools/BuilderWand/components/PremaPanel/inputs/index.tsx:4]: don't need /index

**Review by alex-shortt — APPROVED** [review]
revert adding .tsx to `src/layers/basis/index.tsx`

**Review by alex-shortt — CHANGES_REQUESTED** [review]
sorry accidentally hit approve. just change those last two things and we're good to go

**Review by alex-shortt — APPROVED** [review]
